nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -13,6 +15,134 @@
 PathType = Union[str, Path]
 
 
+class VideoCaptureContext:
+    def __init__(self, *args, stub=False, **kwargs):
+        self.vc = cv2.VideoCapture(*args, **kwargs)
+        self._args = args
+        self._kwargs = kwargs
+        self.stub = stub
+        self._current_frame = 0
+        self.frame_count = self.get_movie_frame_count()
+        assert self.frame_count > 0, "movie contains no frames"
+        self.fps = self.get_movie_fps()
+        self.frame = self.get_movie_frame(0)
+        assert self.frame is not None, "unable to read the movie file provided"
+
+    def get_movie_timestamps(self):
+        """
+        Return numpy array of the timestamps for a movie file.
+
+        """
+        if not self.vc.isOpened():
+            raise ValueError("movie file is not open")
+        ts = [self.vc.get(cv2.CAP_PROP_POS_MSEC)]
+        for i in tqdm(range(1, self.get_movie_frame_count()), desc="retrieving video timestamps"):
+            self.current_frame = i
+            ts.append(self.vc.get(cv2.CAP_PROP_POS_MSEC))
+        self.current_frame = 0
+        return np.array(ts)
+
+    def get_movie_fps(self):
+        """
+        Return the internal frames per second (fps) for a movie file.
+
+        """
+        if int(cv2.__version__.split(".")[0]) < 3:
+            return self.vc.get(cv2.cv.CV_CAP_PROP_FPS)
+        return self.vc.get(cv2.CAP_PROP_FPS)
```

Comment:
```suggestion
    def get_movie_fps(self):
        """Return the internal frames per second (fps) for a movie file."""
        if self.vc.isOpened():
            if int(cv2.__version__.split(".")[0]) < 3:
                return self.vc.get(cv2.cv.CV_CAP_PROP_FPS)
            return self.vc.get(cv2.CAP_PROP_FPS)
```

This could make access to this field even safer.

Reply:
yes, thanks for catching that. I think, better than returning `None` if not `self.vc.isOpened()`is to raise an Exception. 
In fact I am now asserting `isOpened()` for all these methods https://github.com/catalystneuro/nwb-conversion-tools/pull/392/commits/783a2a84331272de0b29da93f862c6661d068046